src/Tooltip/Tooltip.tsx

```diff
@@ -8,37 +12,92 @@ export interface TooltipMessageGroupItem {
     message: string;
 }
 
-interface TooltipState {
-    toggle: boolean;
-}
-
+export type TooltipTrigger = "hover" | "click" | "focus";
 export type TooltipTheme = "default" | "light" | "primary" | "warning" | "success" | "danger" | "purple";
 export type TooltipPosition = "top" | "bottom" | "left" | "right" | "top-right" | "top-left" | "bottom-right" | "bottom-left" | "left-top" | "left-bottom" | "right-top" | "right-bottom";
 
 export interface TooltipProps {
     className?: string;
     customSvg?: React.ReactNode;
     id?: string;
+    ref?: React.LegacyRef<Tooltip>;
+    /** @deprecated use content instead */
     message?: string;
+    /** @deprecated use content instead */
     messageGroup?: Array<TooltipMessageGroupItem>;
-    onClick?: (event?: React.MouseEvent<HTMLDivElement>) => void;
+    /** @deprecated use onVisibleChange instead */
+    onClick?: (event?: React.MouseEvent<HTMLDivElement> | React.FocusEvent<HTMLElement>) => void;
     position?: TooltipPosition;
     theme?: TooltipTheme;
+    /** @deprecated use content instead */
     title?: string;
+    /** @deprecated use trigger instead */
     triggerOnHover?: boolean;
+    /** @deprecated */
     width?: number;
+    content?: string | React.ReactNode;
+    trigger?: TooltipTrigger;
+    disableAutoPosition?: boolean;
+    onVisibleChange?: (event: React.MouseEvent<HTMLDivElement> | React.FocusEvent<HTMLElement>, visible: boolean) => void;
+    children?: React.ReactNode;
+}
+
+interface TooltipState {
+    id: string;
+    visible: boolean;
+    tooltipContainer: HTMLDivElement;
+    tooltipPositionChecker: TooltipPositionChecker;
 }
 
 export class Tooltip extends React.Component<TooltipProps, TooltipState> {
+    private containerRef: React.RefObject<HTMLDivElement> = React.createRef();
+    private tooltipRootClassName: string = "tooltip-root-container";
     constructor(props: TooltipProps) {
         super(props);
 
         this.state = {
-            toggle: false
+            id: "",
+            visible: false,
+            tooltipContainer: null,
+            tooltipPositionChecker: null
         };
 
         this.forceDismiss = this.forceDismiss.bind(this);
-        this.toggleTooltip = this.toggleTooltip.bind(this);
+    }
+
+    componentDidMount() {
+        const randID: string = this.constructId();
```

Comment:
move the logic of constructing the ID here since you never reuse it. And rename `id` in the state to `accessibilityId` or `containerId` or anything else that sets it apart from `this.props.id`

Reply:
renaming state.id to state.accessibilityId